lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -1856,7 +1809,27 @@ impl<'a> FuncGen<'a> {
         self.assembler
             .emit_mov(Size::S64, Location::GPR(GPR::RSP), Location::GPR(GPR::RBP));
 
-        // Initialize locals.
+        // Setup the registers (incl. defining the vmctx register)
+        let local_count = self.local_count();
+        self.machine
+            .setup_registers(&mut self.assembler, local_count, self.signature.params().len() as u32, self.calling_convention);
+
+        // Verify stack height
+        self.assembler.emit_sub(
+            Size::S32,
+            Location::Imm32(self.stack_size),
+            Location::Memory(
+                Machine::get_vmctx_reg(),
+                self.vmoffsets.vmctx_stack_limit_begin() as i32,
+            ),
+        );
+        self.assembler
+            .emit_jmp(Condition::Signed, self.special_labels.stack_overflow);
```

Comment:
Is signed a truly appropriate condition to use here? Consider for example, if `vmctx_stack_limit = 0`, `stack_size = 0xFFFF_FFFF` – after this subtraction we’d end up with a positive number still (`0x1`), so this condition wouldn’t fire even though stack has been overflowed. I think this should be using `Condition::Carry` (CF) which will get set anytime a wrap-around occurred for an unsigned subtraction operation.

Reply:
It’s code carried over from the previous secondary stack check accounting, so I hadn’t checked it in-depth. Thinking about it some more, it does seem to make sense to have all computations use unsigned integers, so I’m applying your suggestion and using `Condition::Carry` :)